data/lua/feeding.lua

```diff
@@ -14,21 +14,33 @@ on_event("die", function()
 	local u_killer = wesnoth.units.get(ec.x2, ec.y2)
 	local u_victim = wesnoth.units.get(ec.x1, ec.y1)
 
-	if not u_killer or not u_killer:matches { ability = "feeding" } then
+	if not u_killer or not u_killer.active_attack then
 		return
 	end
+
+	local feeding = u_killer.active_attack:active_specials {
+		tag = "feeding",
+		base_value = 0,
+	}
+
+	if feeding.value <= 0 then
+		return
+	end
+
 	if not u_victim or u_victim:matches { status = "unplagueable" } then
 		return
 	end
+
+	local text = wesnoth.format(_"+$value max HP", { value = feeding.value})
```

Comment:
This should probably now use wmlxgettext's feature for translatable plurals, and also have a comment for translators.

Reply:
i don't really know how to do that bu i can try

Reply:
i tried but coudn't find a lua api for that.

Reply:
It wasn't documented on the Wiki (now fixed), but comments starting `po:` are supported in the same way as the WML ones.
https://wiki.wesnoth.org/GettextForWesnothDevelopers#Marking_up_strings_in_Lua

Please forget my suggestion about supporting plurals. I thought some languages might need a plural of "HP",  but that seems wrong now.

Reply:
> It wasn't documented on the Wiki (now fixed), but comments starting po: are supported in the same way as the WML ones.

Thx for the help, but i I wasn't talking about the comment, i was talking about the plurals.

Reply:
oh ne3vermind the wiki seems to also explain that :/ , but it looks wrong? it seems to call string.format with a string that expects variable subsitution in wml style.

Reply:
also that wiki page is a little hard to find, i'd expect at least a ling here https://wiki.wesnoth.org/LuaWML/Display#wesnoth.textdomain (which is where i'd look for such lua api)

Reply:
> looks wrong?

It's copied from data/lua/wml/objectives.lua , so I expect it to work.